---
title: "Initial Project Ideas"
author: "Salome Steinke"
date: "26.04.2019"
output:
  word_document: default
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
# Initial Project Ideas

Type of cancer chosen: Breast Cancer 

## The Dataset

Loading Dataset as follows:
```{r Data}
allDepMapData = readRDS("~/4.FS/Bioinfo/DepMap19Q1_allData.RDS")
names(allDepMapData)
```

The list named `allDepMapData` contains the following matrices:

* **Gene expression**: consists of TPM (transcripts per million) values reflecting level of gene expression. Higher values suggest overexpression of genes. Matrix rows are gene names and columns cancer cell line identifiers. Column and row names *character*; TPM values *integer*

```{r Visualising Data1}
dim(allDepMapData$expression)
typeof(col(allDepMapData$expression))
```

* **Gene copy number**: consists of gene copy number (CN) values. When two alleles per gene present CN = 2, deletions shown by CN < 2 and amplifications CN > 2. Matrix rows are gene names and columns cancer cell line identifiers.

* **Gene mutations**: Annotation file for various mutations observed in sample. `isDeleterious` flag specificies if mutation has functional effect or not.

* **Gene knockdown (CERES)**: Gene knockdown scores. These scores measure how essential a particular gene is to cell survival, by reflecting whether a knockdown of the gene reduces , increases or leads to no change of cell proliferation. Smaller values refer to higher importance. Matrix rows are gene names and columns cancer cell line identifiers.

```{r Visualising Data2}
dim(allDepMapData$kd.ceres)
```

* **Gene knockdown (Probability)**: The probabilty for the effect of gene knockdown, with higher probabilities signifying a high likelyhood of the gene reducing cell proliferation. 

```{r Visualising Data3}
dim(allDepMapData$kd.prob)
```

* **Annotation**: This matrix gives information regarding cell lines. The `CCLE_Name` encoded as cell line name _ Tissue of origin. The `DepMap_ID`gives uniform cell line identifiers used in all data sets. The columns  `Primary.Disease` and `Subtype.Disease` refer to tissue/tumor of origin. 

```{r Visualising Data4}
dim(allDepMapData$annotation)
colnames(allDepMapData$annotation)
rownames(allDepMapData$annotation)

```

## Questions so far

1. Which genes are in dataset? 
2. Which genes are we interested in?
3. What are cell line identifiers? Which ones do we have? Perhaps analyse using the *Annotation Matrix*
4. Analyse TPM levels for selected genes and cell line identifiers *(Gene expression matrix)*
5. Analyse *Gene knockdown matrix* to check for importance of genes in cancer cell lines for survival. Compare to *Gene knockdown probability* to check for probability of reducing/increasing cell proliferation.
6. Analyse gene copy numbers for genes and cell line identifiers to identify in which cell lines mutations have occured (compare gene copy numbers for different cell lines) *(Gene copy number matrix)*
7. In cell lines where mutations detected check if they have functional effect or not *(Gene mutations matrix)*
8. which are importat driver mutations in breast cancer? Which are the most important passenger mutations associated with them?
9. ist there an association between driver and passenger alterations? Could synthetic lethal interactions be established by interfering in the interactions between driver and passenger genes?
10. are there any drugs availabe so far targeting the mutations?
11. how could the obtained data be used for cancer therapy?

## Things to do

* identify 3-5 most prominent mutations/genetic alterations (ideally those difficult to target clinically)

* identify appropriate cell line models

* read Papers and extract interesting questions, methods, analysis

* Questions with milestones/ timeframe 

* Data clean up and exploratory

 - Extracting BCCL as dataframe
```{r}
#finding all breast cancer cell lines
BCCL_Numbers <- which(allDepMapData$annotation[4]== "Breast Cancer")
BCCL_Annotation <- subset(allDepMapData$annotation, Primary.Disease == "Breast Cancer")
#see what we got
summary(BCCL_Annotation)
# copy expression matrix to new dataframes
df1 <- allDepMapData$expression
df2 <- allDepMapData$copynumber
df3 <- allDepMapData$kd.ceres
df4 <- allDepMapData$kd.prob
#change column names to numbers for easier handling
names(df1) <- c(1:544)
names(df2) <- c(1:544)
names(df3) <- c(1:544)
names(df4) <- c(1:544)
#extract BCCL as dataframe
BCCL_Expression <- as.data.frame( df1[, c(BCCL_Numbers)])
BCCl_Copynumber <- as.data.frame( df2[, c(BCCL_Numbers)])
BCCL_kd.ceres <- as.data.frame( df3[, c(BCCL_Numbers)])
BCCL_kd.prob <- as.data.frame( df4[, c(BCCL_Numbers)])
dim(BCCL_Expression)
dim(BCCl_Copynumber)
dim(BCCL_kd.ceres)
dim(BCCL_kd.prob)
#reorder rows alphabetically
BCCL_Expression <- BCCL_Expression[order(rownames(BCCL_Expression)), ]
BCCl_Copynumber <- BCCl_Copynumber[order(rownames(BCCl_Copynumber)), ]
BCCL_kd.ceres <- BCCL_kd.ceres[order(rownames(BCCL_kd.ceres)), ]
BCCL_kd.prob <- BCCL_kd.prob[order(rownames(BCCL_kd.prob)), ]
#subset Mutation matrix in BCCL
BCCL_Mutation <- allDepMapData$mutation[c(BCCL_Numbers)]
```
 - include patients ID as first row in BCCL dataframes
```{r echo=FALSE}
Patients_ID <- BCCL_Annotation[1] #get patients ID from annotation matrix
tPatients_ID <- t(Patients_ID) #transpose Patients_ID because it was a column and we want to insert it as row
rownameskdc <- rownames(BCCL_kd.ceres) # save old rownames
existingDF <- BCCL_kd.ceres  # define variables for BCCL_kd.ceres
r <- 1
newrow <- tPatients_ID
#defining a function we can use more often to insert rows
insertRow <- function(existingDF, newrow, r) { 
    existingDF[seq(r+1,nrow(existingDF)+1),] <- existingDF[seq(r,nrow(existingDF)),]
    existingDF[r,] <- newrow
    existingDF
}
BCCL_kd.ceres_ID <- insertRow(existingDF, newrow, r)
rownames(BCCL_kd.ceres_ID) <- c("Deep_ID", rownameskdc) # change rownames back 
BCCL_kd.ceres_ID[1:2, ]
#do the same with other 3 dataframes
rownamesex <- rownames(BCCL_Expression) #save rownames
existingDF <- BCCL_Expression  # define variables for BCCL_Erpression
r <- 1
newrow <- tPatients_ID
BCCL_Expression_ID <- insertRow(existingDF, newrow, r)
rownames(BCCL_Expression_ID) <- c("Deep_ID", rownamesex)
rownamescn <- rownames(BCCl_Copynumber) #save rownames
existingDF <- BCCl_Copynumber  # define variables for BCCL_Copynumber
r <- 1
newrow <- tPatients_ID
BCCl_Copynumber_ID <- insertRow(existingDF, newrow, r)
rownames(BCCl_Copynumber_ID) <- c("Deep_ID", rownamescn)
rownameskdp <- rownames(BCCL_kd.prob) #save rownames
existingDF <- BCCL_kd.prob  # define variables for BCCL_kd.prob
r <- 1
newrow <- tPatients_ID
BCCL_kd.prob_ID <- insertRow(existingDF, newrow, r)
rownames(BCCL_kd.prob_ID) <- c("Deep_ID", rownameskdp)
```
 
 - extracting data for specific driver mutation 
 
```{r}
#find ceres values of MYC mutation
View(BCCL_kd.ceres["MYC", ])
View(BCCL_Expression["MYC",])
View(BCCl_Copynumber["MYC",])
#find cell lines with MYC mutation -> doesn`t look good :) 
```
 

* Data exploration (Descriptive statistics, graphical representations of data and interdependancies )

* Dimensionality reduction (PCA? ; kmeans or clustering of samples)

* find which are most correlated mutations/copy number changes

* functional/biological relationship between correlated alterations

* find clinically approved drugs for correlated alterations

* Data modelling (prediction of cell line sensitivity to knockdown or impact of gene mutation/copy number on gene expression)


# NOTES ON RESEARCH PAPERS
### BREAST CANCER MUTATIONS
### A Comprehensive Pan-Cancer Molecular Study of Gynecologic and Breast Cancers (Berger et al., 2018):

*	Focus on invasive breast carcinoma (BRCA) (1087 Patients)
* Mutations: TP53; PIK3CA; MYC, CCND1
*	ARID1A, ERBB3, BRCA1, FBXW7, KMT2C, PIK3CA, PIK3R1, PPP2R1A, PTEN, and TP53 were mutated at higher frequencies across the Pan-Gyn tumor types than across the non-Gyn types (false discovery rate [FDR] < 0.01, Fisher’s exact test) (Figure 1A)
*	The top ﬁve most frequently mutated genes were TP53 (44% of samples mutated), PIK3CA (32%), PTEN (20%), ARID1A (14%), and PIK3R1 (11%)
*	Correlation coefficient calculations with heat maps and tree diagrams (unsupervised hierarchical clustering) showing clusters 
*	Mutation signatures- unique mutation types – displayed using six substitution subtypes
    +	<https://cancer.sanger.ac.uk/cosmic/signatures>
    +	Breast cancer mutation signatures: Signature 1, signature 3, signature 5 (aetiology unknown), signature 8, signature 17 and signature 18 (aetiology unknown), signature 20, signature 26 
    +	May be interesting to compare identified mutations in genes from own dataset to the substitution types known for breast cancer 
*	Breast cancer associated with COSMIC 5
*	Methods used: 
    + Fisher’s Extact test with Bejamini-Hochberg method for adjusted for false discovery rates
    +	Mann Whitney U test to identify genes with significant differs between meidain methylation levels of genes between two groups
    +	MutSigCV v1.4 <www.broadinstitute.org>
    +	Silhouette analysis 
    +	T-test and Wilcoxon Rank Sum Test 
    +	Survival analysis with R package “survival” 
        -	Kaplan-Meier or Cox Proportional Hazards Models 
        - For more info:  <https://www.datacamp.com/community/tutorials/survival-analysis-R> 

### Comprehensive Molecular Portraits of Invasive Lobular Breast Cancer (Ciriello et al., 2015):

*	817 breast tumor samples
*	Invasive lobular carcinoma (ILC)- second most prevalent histologic subtype of invasive breast cancer (10-15% oof all cases)
    +	Most were identified as Luminal 
*	E-cadherin (CDH1) loss in 90% of cases
    +	Rare HER2 protein overexpression or amplification
    +	Tumor invasion and metastasis in other tumor types including diffuse gastric cancer 
*	ILC, IDC, mixed and other carcinoma types
*	Mutations targeting PTEN, TBX3, FOXA1, PIK3CA, CDH1
    +	PTEN loss associated with increased AKT phosphorylation 
    +	GATA3 mutations- high expression luminal A IDC- differential modulation of ER activity in ILC/IDC
    +	TP53 und PIK3CA in ILC und IDC as well as MYC
    +	ILC cases enriched in CDH1 mutations
    +	Recurrently mutated genes identified by MutSigCV2 
    +	Recurrent copy-number alterations estimated by GISTIC
*	FOXA1
    +	Key ER transcriptional modulator coordinating ER DAN binding within a large protein cmplex by modifying chromatin accessibility and mediating long-range DNA interactions
    +	Mutation cluster in loop 2 of forkhead-DNA (three a-helices, 3 ß-strands and two loops) binding and C-term. transactivation domains 
*	PTEN
    + Homozygous deletions, somatic mutations and mutually exclusive with PIK3CA mutations 
    +	Lower PTEN protein in LumA ILC than LumA IDC
    +	PTEN protein as negative regulator of Akt activity 
*	68 mutated genes, 47 regions of gain, 63 regions of loss identified 
* Genetic difference between ER+/luminal and ER-/basal-like breast cancer 
*	HER2+ breast tumours

### Comprehensive molecular portraits of human breast tumours (The Cancer Genome Atlas et al., 2012) https://doi.org/10.1038/nature11412:

*	TP53, PIK3CA and GATA3 with subtype-associated mutations in GATA3, PIK3CA and MAP3K1 with luminal A subtype
*	Similar aetiology and therapeutic opportunities with ovarian cancer
*	Three groups of breast cancer
    + Oestrogen receptor (ER) positive group most numerous and diverse
    +	HER2 (or ERBB2) amplified group- great clinical success in targeting HER2
    +	Basal-like breast cancers (triple-negative breast cancers) lack expression of ER, progesterone receptor (PR) and HER2
    +	PIK3CA, PTEN, AKT1, TP53, GATA3, CDH1, RB1, MLL3, MAP3K1 and CDKN1B
    +	With novel mutations: TBX3, RUNX1, CBFB, AFF2, PIK3R1, PTPN22, PTPRD, NF1, SF3B1 and CCND3
    +	FOXA1 and CTCF transcription factor mutations
    + The luminal A subtype harboured the most significantly mutated genes, with the most frequent being PIK3CA (45%), followed by MAP3K1, GATA3, TP53, CDH1 and MAP2K4. (missense)
    +	Luminal B cancers : TP53 and PIK3CA (29% each) most frequent (missense)
    +	Basal like cancers: TP53 (nonsense, frame shift)
    + HER2E subtype: high frequency of TP53 (72%) and PIK3CA (39%) mutations + lower frequency with PIK3R1 (4%)
* Luminal/ER+: 
    +	ESR1, GATA3, FOXA1, XBP1 and MYB
    +	GATA3 and FOXA1 mutated in mutually exclusive fashion 
    + PIK3CA high mutation frequency 
    +	Frequent MAP3K1 and MAP2K4 mutation 
    + MEMo analysis: all MAP3K1 and MAP2K4 mutations were in luminal tumors (almost mutually exclusive to each other though
    +	TP53 highest in luminal A cancers
    + Hyperactivation transcriptional activity with MYC and FOXMI proliferation 
    +	Luminal A tumours- retain RB1 and TP53 activity
* BRCA1/BRCA2
*	Basel-like cancers
*	TP53, RB1 and BRCA1 mutations  

- notes from elias, just to not erase them :)
* Methods used were genomic DNA copy number arrays, DNA methylation, exome sequencing, messenger RNA arrays,
  microRNA sequencing and reverse-phase protein arrays 
*	The analysis revealed 4 major breast cancer classes, which are different in gene expression
*	In therapy 3 major classes are distinguished: ER+ Patients, which receive endocrine therapy, 
  HER2 amplification patients (we had strategies treatment strategies in lecture) and triple negative breast cancer (TNBC) patients,
  often BRCA1 is mutated, patients can only be treated with chemotherapy
*	The new step of this paper is to include the methods mentioned above. Earlier studies just used data from mRNA expression,
  DNA copies and parallel sequencing (no idea what the last thing is)
*	Total number of patients is 825
*	A LOT  of different mutations were found during whole exome sequencing, which lead to following new genes associated with breast cancer:
  TBX3, RUNX1, CBFB, AFF2, PIK3R1, PTPN22, PTPRD, NF1, SF3B1 and CCND3.
*	Genes confirmed and already known to be important for breast cancer development were:
  PIK3CA, PTEN, AKT1, TP53, GATA3, CDH1, RB1, MLL3, MAP3K1 and CDKN1B.
*	mRNA-expression of cancer subtypes revealed different mutation patterns in different subclasses ( needs a more detailed description)
*	Patterns vary also by mutation type (nonsense, misssense etc.)
*	Further analysis showed a strong connection to the germ line of ~10% of breast cancers, including BRCA1 mutations,
  confirming the TNBC subclass (not the best description I ever made 😊)
  
  ###Genetic Interactions in Cancer Progression and Treatment (Ashworth et al.2011) DOI: 10.1016/j.cell.2011.03.020
  
### Predicting Cancer-Speciﬁc Vulnerability via Data-Driven Detection of Synthetic Lethality:

*	Synthetic lethality: inhibition of two nonessential genes lethal while inhibition of a single gene is not
*	Targeting synthetic lethal (SL) partners in treatment
*	Use of DAISY (data mining synthetic lethality identification pipeline)
    ++	Genomic survival of the fittest (SoF) based on observation that cancer cells having lost two SL-paried genes don’t survive – SL cells elimination form population hence somatic copy number alterations (SCNA) and somatic mutation data can detect this inactivation events
    ++	SL pairs of given gene can be detected underexpression or low copy number induces essentiality of the gene
    ++	Pairwise gene coexpression 
    ++	Gene pairs fulfilling all three of the above 
    ++	Detection of overactivity also possible by selecting two genes if expression is correlated and if overactivity of one induces essentiality of the other (first examining if gene B becomes essential when gene A is overactive; second if gene B has a higher SCNA level when gene A is overactive)
    ++	Returns p values denoting significance of SL or SDL interaction
*	Wilcoxon rank sum p values 
*	Predicting gene essentiality is cell-line-specific 
*	What are ROC curves? 
*	Signed Kaplan-Meier (KM) score 
*	Experimental procedures section most interesting for us:
    ++	(1) Wilcoxon rank sum test to determine of gene B significantly higher SCNA levels when gene A is inactive/overactive in rest of samples. Ouput= gene pairs that pass the test with specific significant level
    ++	(2) input: gene essentiality data with SCNA and gene expression profiles. Wilcoxon rank sum test to determine if Gene B lower scores where gene A inactive/overactive 
    ++	P value < 0.05 used with Bonferroni correction 
    ++	Gene defined underexpressed if SCAN below -0.3; likewise overactive when above 0.3 
    ++	Generally genes in sample underexpressed if in 10th percentile of all expression levels and overexpressed if above the 90the percentile 
    ++ For clustered genes and gene maps visit < http://www.cs.tau.ac.il/livnatje/SL_network.zip >

### Review: Synthetic lethality and cancer (O'Neill, 2017):
    
 * **synthetic lethal interaction (SLI)**: interaction of two genes which is only lethal if both genes are perturbated BUT NOT if only one gene is perturbated  
    ++ involves cancer-specific interaction which is "non-oncogene addiction" + one tumour-specific somatic mutation
    ++ mutations may contribute both to tumour proliferation and/or an increase in tumour vulnerability to tgerapeutics
    ++ therapeutical concept: targeting of the passenger mutation (= second-site synthetic lethal target) & overcoming undruggable "oncogene addiction" mutations
 * **challenges in identfying SLI**:
    ++ SLI may be condition-dependend (both refering to genetic background & cellular conditions)
    ++ SLI are rare
    ++ identification of mutants difficult as cells die due to SLI which impedes mutant recovery
 * **types of synthetic lethality**:
    ++ synthetic dosage lethality (SDL): overexpression of one gene (due to epigenetic gain-of-function mutation or increase in gene copy number) interacting with another gene reduced in function results in lethality (breast cancer: SDL of high CKS1B with inhibition/knockdown of PLK1) 
    ++ conditional synthetic lethality: intrinisic (genetic) and extrinsinc (tumour microenvironment, i.e. hypoxia) may be predispositions for efficiency of SLI, thus alteration of tumour-specific conditions may be useful in order to increase cytotoxicity of therapeutics (breast cancer: use of PARP-inhibitors in homologous recombination deficient tumours, also SLI of PARP with BRCA1)
 * **screening for SLI in model organisms**:
    ++ use of high-throughput approaches in yeast based on existing model organis interaction networks for SLI with cancer-relevant homologues, if no homologues available then screening for cancer-relevant pathways & disruption of gene involved in a similar process in humans
    ++ use of synthetic lethality genetic networks in yeast: network based on quantification of digenic interactions generated by double mutant construction -> identification of > 500.00 SLIs, findings may be used for identifiation of human (homologue & non-homologue) SLI candidates due to evolutionary conservation  
    ++ SLI screening in human cells: CRISPR outperforms RNAi due to greater RNA libraries & stability of CRISPR based gene knockouts, BUT cell line specificty & targeting (RNAi: due to interference with endogenous RNAi pathways, CRISPR: due to off-target effects & toxicity in aneuploid cells) remain problematic 
        -> use of CRISPR gene editing, CRISPRi & CRISPRa (fusion with repressor/activator), CombiGEM-CRISPR (multiplexed sgRNAs targeting up to 6 sequences), Pertub-seq (CRISPR + single-cell transriptome analysis instead of effects on cellular fitness)
    ++ predicting SLI candidates based on statistical genetics: search for mutually exclusive (= SLI suitable) mutations in databases (e.g. The Cancer Genome Atlas, Calatolgue of Somatic Mutations in Cancer) & use of DAISY (bioinformatic approach using genes essentiality profiles, expression data & data on copy number alterations)
  * **use of SLI for cancer therapy**:
    ++ SLI reported in SLI databases (e.g. SynLethDB)
    ++ therapeutic approaches may involve increasing the tumour susceptibility to conventional treatments (i.e. chemotherapy) & exploiting SLI by manipulation of the second-side targets (passenger mutation, e.g. PARP-inhibitot in breast cancer)
    ++ HOWEVER resistances & off-target effects have yet been poorly studied 

### Toward an integrated map of genetic interactions in cancer cells (Rauscher et al.2018 )DOI: 10.15252/msb.20177656

* As basics for a new modelling method:
  + Previous description: statistical epitasis, defined as statistical deviation from the additive combination of two (gene) loci and how it affects the phenotype of interest
  + problematic, if data derives from various libraries or experimental protocols
  + new MINGLE method leads to interaction map sorted into genes with similar profiles and network molecules with similar functions
* Main Steps of the proposed method
    + available on <https://github.com/boutroslab/Supplemental-Material/tree/master/Rauscher_2017>

* Methodic elements of the model
    + considered alterations: gain or loss of copy events, somatic mutations and mRNA overexpression
    + gene-level correction and library batch effect determined with regression model with beta=0 and yi = y(crispr,i) - median(crispr,i) and a F-test (R-package sfsmisc) if the median crispr score is the same over all libraries (H0). Batch effect was defined as "phenotypes generated by one library differing significantly (FDR < 5%) from the observed median phenotype across all libraries"
    + control of normalization by Ward clustering/heatmap (hierachical clustering, do biologically related samples cluster closer than unrelated?)
    + comparison of control oncogenes towards gene knockout in the different libraries (2-sided student´s t-test); result was compared with literature
    + quantification of geneteic interactions with pi-score statistic
    + combinatorial testing of gene-gene-interactions: mixed model of regression analysis (fixed: alteration genotype, random: cell line)
    + gene similiratity network benchmarking and modeling (with pi-score and spearman; wasn´t able to understand the process completely)

* Key points of genetic interactions :
    + GANAB and PRKCSH as positive regulators of Wnt/beta-catenin signaling
    + RNF43 as negative regulator of the signaling pathway 
 

>References: 

>Berger, A.C., Korkut, A., Kanchi, R.S., Hegde, A.M., Lenoir, W., Liu, W., Liu, Y., Fan, H., Shen, H., Ravikumar, V., et al. (2018). A Comprehensive Pan-Cancer Molecular Study of Gynecologic and Breast Cancers. Cancer Cell 33, 690-705.e699.
>Ciriello, G., Gatza, Michael L., Beck, Andrew H., Wilkerson, Matthew D., Rhie, Suhn K., Pastore, A., Zhang, H., McLellan, M., Yau, C., Kandoth, C., et al. (2015). Comprehensive Molecular Portraits of Invasive Lobular Breast Cancer. Cell 163, 506-519.
>The Cancer Genome Atlas, N., Koboldt, D.C., Fulton, R.S., McLellan, M.D., Schmidt, H., Kalicki-Veizer, J., McMichael, J.F., Fulton, L.L., Dooling, D.J., Ding, L., et al. (2012). Comprehensive molecular portraits of human breast tumours. Nature 490, 61.
Ashworth, A., Lord, C.J. & Reis-Filho, J.S., 2011. Genetic Interactions in Cancer Progression and Treatment. Cell, 145(1), S.30–38
>Jerby-Arnon, Livnat, et al. “Predicting cancer-specific vulnerability via data-driven detection of synthetic lethality.” Cell 158.5 (2014): 1199-120

### Possible Mutations for analysis: 

1.	PTEN: tumor suppressor gene which is silenced by the means of point mutations or promotor methylation in aprox. 42.5 % of breast cancer patients     (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3742525/), function of protein and lipid phosphatase with role in apoptosis & negative regulation of proliferation (https://www.uniprot.org/uniprot/P60484)
      -> if we are looking for overexpressed driver mutations PTEN might be unsuitable

2.	TP53

3.	PIK3CA: 
      

4.	MYC 
    has high copynumber in BCCL_Copynumber sometimes, but i couldn`t find mutations in a quick manual view

5.	GATA3: role in cancer progression stil is not clear, especially, there seeems to be no correlation between expression levels and tumour aggressiveness. transription factor with crucial role in cell differentiation (https://academic.oup.com/ajcp/article/144/5/756/1761258)

6.	BRCA1/BRCA2: cumulative cancer risk for mutation carriers (deletions, premature Stop codon, missense) until age of 80 years 72% and 69%, respectively (https://jamanetwork.com/journals/jama/fullarticle/2632503, https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4537700/), BRCA1 function: E3 ubiquitin-protein ligase with role in tumor suppression, BRCA2 function:involved in DSB repair & homologous recombination (https://www.uniprot.org/uniprot/P51587)
      -> again: these driver mutations do not lead to overexpression

7.	RB1: tumor suppressor gene which is typically downregulated or completely lost in cancer cells (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3055199/), promotor of G0-G1 transition in cell cycle (https://www.uniprot.org/uniprot/P06400)
      -> HOWEVER RB1 is thought to be downregulated by phosphorylation by CCND1 which is upregulated in 50% of breast cancers 

8.	PARP

9. 	GANAB

10.	PRKCSH

11.	RNF43

12. CDH1: encodes E-cadherin protein which has a role in cell adhesion, increase in cell mobility and metastasis, is expressed normally in many breast cancer cells as mutations mainly concern the extracellular domain and, thus, decrease adhesion to ECM and enable metastases (https://link.springer.com/article/10.1007/s10689-016-9869-5)

```{r echo=TRUE}
tBCCL_Expression <- t(BCCL_Expression)
boxplot(tBCCL_Expression[, c("MYC", "PTEN", "TP53", "PIK3CA", "GATA3", "BRCA1", "BRCA2", "RB1", "GANAB", "PRKCSH", "RNF43")], xlab = "Suggested driver mutations", ylab ="Transcripts per million")

```

### Meeting 1: Friday, 03/05/2019

* **main problem: no expression data for healthy cells:** 
  ++ no specification of alterations in gene expression in cancer cells compared to healty cells
  ++ no specification possible if knockdown of a gene which has a deleterious effect on cancer cells does not have a deleterious effect on heathy cells
      -> solution: rely on information about cancer-specific mutations given in papers
  
* **strategy for overexpression (both):**
  ++ identify driver mutations in papers
  ++ identfy cell lines with certain driver mutations
  ++ get ceres score for driver mutation (negative!) & correlate with all other ceres values of selected cell lines
      -> assumption: second-site targets should also have negative ceres scores!!!
      -> conclusion: high correlation between ceres scores of interacting mutations (high positives)
  ++ in mutation matrix for selected cell lines: look for mutation of second-site targets
  
* **strategy for overespression of driver gene and underexpression of second-site target:**
  ++ identify driver mutations in paper
  ++ identfy cell lines with certain driver mutations
  ++ correlate expression data (TPM) of driver mutations (high TPM) with data of all other genes in selected cell lines
      -> assumption: low expression profiles for second-site targets
      -> conclusion: high negative correlation values 
  ++ in ceres matrix: look for ceres scores of second-site targets 
      -> ceres scores of second-site mutations supposed to be 0 (gene knockout ~ underexpression)
  ++ select cell lines with overpressed driver mutation & non-underexpressed second-site target & get ceres scores for driver mutations
      -> assumption: ceres score for driver mutation (non-underexpressed second-site target) > ceres score for diver mutation (overexpressed second-site target) (reason: background of underxpressed second-side target enhances cell viability & proliferation )
      
### Meeting 2: Friday... with David

* **visualize gene expression data from different samples:**
  ++ create boxplot for expression data of different samples & align them for comparison
  ++ data has to be transformed in order to fit boxplots -> find advice online
  ++ R package: GGPlot2 (recommended by David, helpful desriptions & cheat sheets online)
       -> done above :)
  
```{r install ggplot2}
# requires installation of latest Rtool version: https://cran.r-project.org/bin/windows/Rtools/ 
# install package ggplot2
install.packages("ggplot2")
# load package
library("ggplot2")
```
  
* **linear regression to make assumptions regarding ceres scores:**
  ++ x-axis: TPMs; y-axis: ceres values 
      -> do ceres scores correlate with expression patterns of driver mutations/second-site targets?
      
* **check how relevant driver mutations are for cancer cells using PCA:**
  ++ assumption: driver mutations should contribute much to the overall variance
  ++ maybe also check contribution of second-site targets to PCs, HOWEVER impact of second-site targets could be smaller as they are non-oncogenic and only act as "co-effectors" regarding the proliferation of tumour cells
  ++ which data set to use? -> suggestion: ceres scores as both driver mutations & second-site mutations should contribute to a greater extend here, otherwise we coud use the TPM matrix (but again normalization with reference to "healthy" gene expression patterns may be problematic)
  ++ add-on: check whether there are clusters of cancer samples in which an increase in proliferation is driven by one or two main mutations
  
